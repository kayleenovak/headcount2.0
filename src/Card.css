.school-card {
  border-radius: 10px;
  box-shadow: 2px 2px 5px black;
  padding: 10px;
  background: white;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.school {
  font-size: 1.2em;
  pointer-events: none;
  margin: 0 0 20px 0;
}

.greater-than {
 color: blue;
 pointer-events: none;
}

.less-than {
  color: red;
  pointer-events: none;
}

.selected {
  border: 4px solid grey;
  border-radius: 10px;
  box-shadow: 2px 2px 5px black;
  padding: 10px;
  background: white;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.unselected {
  border-radius: 10px;
  box-shadow: 2px 2px 5px black;
  padding: 10px;
  background: white;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}