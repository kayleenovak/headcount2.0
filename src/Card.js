import React, { Component } from 'react';
import './Card.css';
import'./CardContainer.css'

const Card = (props) => {
  const yearStats = Object.keys(props.yearlyStats).map(year => {
    return <li className={props.yearlyStats[year] >= 1 ? 'greater-than' : 'less-than'} key={ year }>{ year }: { props.yearlyStats[year] }</li>
  })

  return(
    <div className='school-card' id={ props.school } onClick={ props.setComparedSchools }>
      <h2 className='school'>{ props.school }</h2>
        { yearStats }
    </div>
  )

}

export default Card