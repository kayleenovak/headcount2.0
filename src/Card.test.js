import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should have a class name of less than if the stat is less than 1'), () => {
    const wrapper = shallow(<Card key='school' yearlyStats={{'2004': .3}} school='BEAR CREEK'/>)

    expect(wrapper.find('li').hasClass('less-than')).toEqual(true)
  }

  it('should have a class name of less than is the stat is greater than 1'), () => {
    const wrapper = shallow(<Card key='school' yearlyStats={{'2004': 1}} school='BEAR CREEK'/>)

    expect(wrapper.find('li').hasClass('greater-than')).toEqual(true)
  }

  it('should return 1 child node', () => {
    const wrapper = shallow(<Card key='BEAR CREEK' yearlyStats={{'2004': .3}} school='BEAR CREEK' />)

    expect(wrapper.find('li').length).toEqual(1)
  })

  it('should match the snapshot with all the data passed in correctly', () => {
    const wrapper = shallow(<Card key='BEAR CREEK' yearlyStats={{'2004': .3}} school='BEAR CREEK' />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should fire the setComparedSchools function on click', () => {
    const mockFunction = jest.fn()
    const wrapper = shallow(<Card key='BEAR CREEK' yearlyStats={{'2004': .3}} school='BEAR CREEK' setComparedSchools={ mockFunction }/>)

    wrapper.find('.school-card').simulate('click')
    expect(mockFunction).toHaveBeenCalled()
  })

})