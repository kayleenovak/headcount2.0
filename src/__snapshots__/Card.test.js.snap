// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all the data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    school="BEAR CREEK"
    yearlyStats={
        Object {
            "2004": 0.3,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="school"
>
          BEAR CREEK
</h2>,
        Array [
          <li
            className="less-than"
>
            2004
            : 
            0.3
</li>,
        ],
      ],
      "className": "school-card",
      "id": "BEAR CREEK",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BEAR CREEK",
          "className": "school",
        },
        "ref": null,
        "rendered": "BEAR CREEK",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "2004",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "2004",
            ": ",
            0.3,
          ],
          "className": "less-than",
        },
        "ref": null,
        "rendered": Array [
          "2004",
          ": ",
          0.3,
        ],
        "type": "li",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="school"
>
            BEAR CREEK
</h2>,
          Array [
            <li
              className="less-than"
>
              2004
              : 
              0.3
</li>,
          ],
        ],
        "className": "school-card",
        "id": "BEAR CREEK",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BEAR CREEK",
            "className": "school",
          },
          "ref": null,
          "rendered": "BEAR CREEK",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "2004",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "2004",
              ": ",
              0.3,
            ],
            "className": "less-than",
          },
          "ref": null,
          "rendered": Array [
            "2004",
            ": ",
            0.3,
          ],
          "type": "li",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
